catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/ScriptFragmentTest.kt

```diff
@@ -0,0 +1,273 @@
+
+        onBrickAtPosition(1)
+            .onCheckBox().perform(click())
+
+        onBrickAtPosition(0)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(1)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(2)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(3)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(4)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(5)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+
+        onBrickAtPosition(5)
+            .checkShowsText(R.string.brick_if_end)
+
+        onBrickAtPosition(5)
+            .onCheckBox().perform(click())
+
+        onBrickAtPosition(0)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(1)
+            .onCheckBox().check(matches(allOf(isChecked(), isEnabled())))
+        onBrickAtPosition(2)
+            .onCheckBox().check(matches(allOf(isChecked(), not(isEnabled()))))
+        onBrickAtPosition(3)
+            .onCheckBox().check(matches(allOf(isChecked(), not(isEnabled()))))
+        onBrickAtPosition(4)
+            .onCheckBox().check(matches(allOf(isChecked(), not(isEnabled()))))
+        onBrickAtPosition(5)
+            .onCheckBox().check(matches(allOf(isChecked(), not(isEnabled()))))
+
+        onBrickAtPosition(1)
+            .onCheckBox().perform(click())
+
+        onBrickAtPosition(0)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(1)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(2)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(3)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(4)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+        onBrickAtPosition(5)
+            .onCheckBox().check(matches(allOf(not(isChecked()), isEnabled())))
+    }
+
+    private fun createProject() {
+        val projectName = javaClass.simpleName
+        val project = Project(ApplicationProvider.getApplicationContext(), projectName)
+
+        val sprite = Sprite("testSprite")
+        project.defaultScene.addSprite(sprite)
+
+        val startScript = StartScript()
+        val ifBrick = IfLogicBeginBrick()
+        ifBrick.addBrickToIfBranch(SetXBrick())
+        ifBrick.addBrickToElseBranch(ChangeXByNBrick())
+        startScript.addBrick(ifBrick)
+        startScript.setParents()
+
+        sprite.addScript(startScript)
+
+        projectManager.currentProject = project
+        projectManager.currentSprite = sprite
+    }
```

Comment:
Use UiTestUtils for a cleaner and easier project creation.
```suggestion
    private fun createProject() {
        val startScript = UiTestUtils.createProjectAndGetStartScript(javaClass.simpleName)
        val ifBrick = IfLogicBeginBrick()
        ifBrick.addBrickToIfBranch(SetXBrick())
        ifBrick.addBrickToElseBranch(ChangeXByNBrick())
        startScript.addBrick(ifBrick)
        startScript.setParents()
    }
```